lib/modules/platform/pr-body.ts

```diff
@@ -21,21 +27,30 @@ export function hashBody(body: string | undefined): string {
   return result;
 }
 
-export function isRebaseRequested(body: string | undefined): boolean {
+function isRebaseRequested(body: string | undefined): boolean {
   return !!body?.includes(`- [x] <!-- rebase-check -->`);
 }
 
+export function getRenovateDebugPayload(body: string): string | undefined {
+  const match = prDebugDataRe.exec(body);
+  return match?.groups?.payload;
+}
+
 export function getPrBodyStruct(
   input: string | undefined | null
 ): PrBodyStruct {
-  const str = input ?? '';
-  const hash = hashBody(str);
+  const body = input ?? '';
+  const hash = hashBody(body);
   const result: PrBodyStruct = { hash };
 
-  const rebaseRequested = isRebaseRequested(str);
+  const rebaseRequested = isRebaseRequested(body);
   if (rebaseRequested) {
     result.rebaseRequested = rebaseRequested;
   }
 
+  const debugPayload = getRenovateDebugPayload(body);
+  if (debugPayload) {
+    result.debugData = JSON.parse(fromBase64(debugPayload));
```

Comment:
Let's say someone edits the base64 either intentionally or accidentally. Won't this then throw and make processing/fixing the PR impossible? I think we should try/catch and warn if so but treat the debug data as non-existing/lost if it won't parse

Reply:
there's no logger in this file should i add?